src/app/mesh/qgsmaptooleditmeshframe.cpp

```diff
@@ -203,6 +211,56 @@ void QgsMeshEditForceByLineAction::updateSettings()
   settings.setEnumValue( QStringLiteral( "UI/Mesh/ForceByLineToleranceUnit" ), mUnitSelecionWidget->unit() );
 }
 
+//
+// QgsMeshEditDigitizingAction
+//
+
+QgsMeshEditDigitizingAction::QgsMeshEditDigitizingAction( QObject *parent )
+  : QWidgetAction( parent )
+{
+  QGridLayout *gLayout = new QGridLayout();
+  gLayout->setContentsMargins( 3, 2, 3, 2 );
+
+  QgsSettings settings;
+
+  QLabel *labelZValueType = new QLabel( tr( "Z value from" ) );
+  mComboZValueType = new QComboBox();
+  mComboZValueType->addItem( tr( "Z Widget" ), ZWidget );
+  mComboZValueType->addItem( tr( "Terrain" ), Terrain );
+  mComboZValueType->addItem( tr( "In Mesh Use Mesh Otherwise Terrain" ), InMeshUseMeshTerrainOtherwise );
+  mComboZValueType->addItem( tr( "In Mesh Use Mesh Otherwise Z Widget" ), InMeshUseMeshZWidgetOtherwise );
```

Comment:
I suggest a slight reword, tooltips and reorder so that current default is first:

```suggestion
  QLabel *labelZValueType = new QLabel( tr( "New vertex Z value" ) );
  labelZValueType->setToolTip( tr( "Sets the source of Z values when adding new vertices" ) );
  mComboZValueType = new QComboBox();
    mComboZValueType->addItem( tr( "Prefer mesh, then Z Widget" ), PreferMeshThenZWidget );
  mComboZValueType->setItemData( 0, tr( "Get interpolated value from the existing mesh.\nIf vertex lies outside the mesh, get the value from the Z value widget instead." ), Qt::ToolTipRole );
    mComboZValueType->addItem( tr( "Prefer mesh, then terrain" ), PreferMeshThenTerrain );
  mComboZValueType->setItemData( 1, tr( "Get interpolated value from the existing mesh.\nIf vertex lies outside the mesh, get the value from the project terrain instead." ), Qt::ToolTipRole );
  mComboZValueType->addItem( tr( "Project terrain" ), Terrain );
  mComboZValueType->setItemData( 2, tr( "Get the value from the project's terrain.\nIf the terrain has no value for the point, the default elevation will be used." ), Qt::ToolTipRole );
  mComboZValueType->addItem( tr( "Z Widget" ), ZWidget );
  mComboZValueType->setItemData( 3, tr( "Always use the value set in the Z value widget" ), Qt::ToolTipRole );
```